x/concentrated-liquidity/store_test.go

```diff
@@ -184,3 +205,113 @@ func (s *KeeperTestSuite) TestParseFullPositionFromBytes() {
+		val           []byte
+		expectedValue genesis.FullTick
+		expectedErr   error
+	}{
+		"valid positive tick": {
+			key:           types.KeyTick(defaultPoolId, defaultTickIndex),
+			val:           cdc.MustMarshal(&defaultTickInfo),
+			expectedValue: defaultTick,
+		},
+		"valid zero tick": {
+			key:           types.KeyTick(defaultPoolId, 0),
+			val:           cdc.MustMarshal(&defaultTickInfo),
+			expectedValue: withTickIndex(defaultTick, 0),
+		},
+		"valid negative tick": {
+			key:           types.KeyTick(defaultPoolId, -1),
+			val:           cdc.MustMarshal(&defaultTickInfo),
+			expectedValue: withTickIndex(defaultTick, -1),
+		},
+		"valid negative tick large": {
+			key:           types.KeyTick(defaultPoolId, -200),
+			val:           cdc.MustMarshal(&defaultTickInfo),
+			expectedValue: withTickIndex(defaultTick, -200),
+		},
+		"empty key": {
+			key:         []byte{},
+			val:         cdc.MustMarshal(&defaultTickInfo),
+			expectedErr: types.ErrKeyNotFound,
+		},
+		"random key": {
+			key: []byte{112, 12, 14, 4, 5},
+			val: cdc.MustMarshal(&defaultTickInfo),
+			expectedErr: types.InvalidTickKeyByteLengthError{
+				Length: 5,
+			},
+		},
+		"using not full key (wrong key)": {
+			key: types.KeyTickPrefixByPoolId(defaultPoolId),
+			val: cdc.MustMarshal(&defaultTickInfo),
+			expectedErr: types.InvalidTickKeyByteLengthError{
+				Length: len(types.TickPrefix) + cl.Uint64Bytes,
+			},
+		},
+		"invalid prefix key": {
+			key:         formatFullKey(types.PositionPrefix, sdk.Uint64ToBigEndian(defaultPoolId), types.TickIndexToBytes(defaultTickIndex)),
+			val:         cdc.MustMarshal(&defaultTickInfo),
+			expectedErr: types.InvalidPrefixError{Actual: string(types.PositionPrefix), Expected: string(types.TickPrefix)},
+		},
+		"invalid value": {
+			key:         types.KeyTick(defaultPoolId, defaultTickIndex),
+			val:         cdc.MustMarshal(&defaultTick), // should be tick info
+			expectedErr: types.ErrValueParse,
+		},
+		"invalid tick index encoding": {
+			// must use types.TickIndexToBytes() on tick index for correct encoding.
+			key: formatFullKey(types.TickPrefix, sdk.Uint64ToBigEndian(defaultPoolId), sdk.Uint64ToBigEndian(defaultTickIndex)),
+			val: cdc.MustMarshal(&defaultTickInfo),
+			expectedErr: types.InvalidTickKeyByteLengthError{
+				Length: len(types.TickPrefix) + cl.Uint64Bytes + cl.Uint64Bytes,
+			},
+		},
+		"invalid pool id encoding": {
+			// format 1 byte.
+			key: formatFullKey(types.TickPrefix, []byte(fmt.Sprintf("%x", defaultPoolId)), types.TickIndexToBytes(defaultTickIndex)),
+			val: cdc.MustMarshal(&defaultTickInfo),
+			expectedErr: types.InvalidTickKeyByteLengthError{
+				Length: len(types.TickPrefix) + 2 + cl.Uint64Bytes,
+			},
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		s.Run(name, func() {
+
+			fullTick, err := cl.ParseFullTickFromBytes(tc.key, tc.val)
+			if tc.expectedErr != nil {
+				s.Require().Error(err)
+				s.Require().ErrorIs(err, tc.expectedErr)
```

Comment:
Can we change this to ErrorContains? ErrorIs has weird behavior that have cause false negatives in the past.

Reply:
What were the problems?

`ErrorIs` is the correct way in this case because we want to check the presence of the `tc.expectedErr` in `err`'s chain.

`ErrorContains` is less preferred because it requires hardcoding an error as a string. So that if an error message changes, we have to update all tests as well. On the other hand, with `ErrorIs` check, the tests can continue working despite a change in the message

Reply:
The problem was that the ErrorIs check was passing despite the error being a completely different error. This has occurred for me on more than one occasion. I can specifically point it out the next time I occur it.

Reply:
This might happen if slices are used as error fields. Since slices reference different buffers under the hood, the equivalence check would fail. From my experience, this frequently happens with byte slices, which I would convert to strings instead.

Reply:
Going to merge but happy to continue this discussion separately and revisit if needed